tern/analyze/docker/analyze.py

```diff
@@ -84,17 +84,21 @@ def abort_analysis():
 
 
 def analyze_first_layer(image_obj, master_list, redo):
-    # find the binary and shell by mounting the base layer
-    target = rootfs.mount_base_layer(image_obj.layers[0].tar_file)
-    binary = common.get_base_bin()
-    shell = get_shell(image_obj, binary)
-    # set up a notice origin for the first layer
-    origin_first_layer = 'Layer: ' + image_obj.layers[0].fs_hash[:10]
-    # only extract packages if there is a known binary and the layer is not
-    # cached
-    if binary:
-        if not common.load_from_cache(image_obj.layers[0], redo):
-            # Determine pacakge/os style from binary in the image layer
+    # set shell to empty string in case one is found
+    shell = ''
+    # try to load from cache
+    loaded = common.load_from_cache(image_obj.layers[0], redo)
+    # if there is no cache, find the binary and shell by mounting the
+    # base layer
+    if not loaded:
```

Comment:
Will this cache have the binary information as well?

Reply:
I guess the cache will have the `/etc/os-release` information so we don't necessarily need the binary.